internal/cmd/sst/cmd.go

```diff
@@ -12,75 +12,77 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package main
+package sst
 
 import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/serverless-sample-tester/internal/sample"
+	"github.com/GoogleCloudPlatform/serverless-sample-tester/internal/util"
 	"github.com/spf13/cobra"
 	"log"
-	"os"
 	"path/filepath"
 )
 
-var (
-	sample *Sample
-
-	sampleDir          string
-	keepContainerImage bool
-
-	allTestsPassed bool
-)
-
-func main() {
-	rootCmd := &cobra.Command{
+// NewCommand creates the root command.
+func NewCommand() *cobra.Command {
+	return &cobra.Command{
 		Use:   "sst [sample-dir]",
 		Short: "An end-to-end tester for GCP samples",
 		Args:  cobra.ExactArgs(1),
-		Run:   root,
-	}
-
-	if err := rootCmd.Execute(); err != nil {
-		log.Panicf("Error with cobra rootCmd Execution: %v\n", err)
-	}
-
-	if allTestsPassed {
-		os.Exit(0)
-	} else {
-		os.Exit(1)
+		RunE:  root,
 	}
 }
 
-func root(cmd *cobra.Command, args []string) {
+// root is responsible for the root command. It handles the application flow.
+func root(cmd *cobra.Command, args []string) error {
 	// Parse sample directory from command line argument
-	var err error
-	sampleDir, err = filepath.Abs(filepath.Dir(args[0]))
+	sampleDir, err := filepath.Abs(filepath.Dir(args[0]))
 	if err != nil {
-		log.Fatalf("Error parsing sample direcotry: %v\n", err)
+		return err
```

Comment:
You can wrap these errors to indicate where they are being thrown from-- this helps with debugging. 

i.e. `return fmt.Errorf("filepath.Abs: %v", err)`

Same for errors below in this file.